### PR TITLE
Enable all metrics by default

### PR DESCRIPTION
**Pull request by RassK**
## Why

Part of https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/388
Follow up of https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/932

## What

* Enables all metrics instrumentations by default
* Checks for ASP.NET Core framework presence

## Tests

Removed `OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS` config from all of the integration tests.

## Checklist

- [x] `CHANGELOG.md` is updated.
- [x] Documentation is updated.
- [x] New features are covered by tests.

### CODE REVIEW

**Review by Kielek — APPROVED** [review]

File: src/OpenTelemetry.AutoInstrumentation/Util/AssemblyDetector.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+        public static bool IsAspNetCoreDetected => _isAspNetCoreDetected ??= DetectAspNetCore();
+
+        private static bool DetectAspNetCore()
+        {
+            return Assembly.GetEntryAssembly()?
+                .GetReferencedAssemblies()
+                .Any(x => x.Name == "Microsoft.AspNetCore") ?? false;
+        }
+    }
```
[the remark this answers is not in the document]
Comment (RassK) [reply src/OpenTelemetry.AutoInstrumentation/Util/AssemblyDetector.cs:34]: Currently it's just a bool stored there. Non-static would mess up the code, cause we have static extensions etc where to pass the reference.

File: test/IntegrationTests/Helpers/TestHelper.cs [hunk 2]
```diff
@@ -90,7 +90,11 @@ public async Task<Container> StartContainerAsync(TestSettings testSettings, int
             Output.WriteLine($"Otlp Endpoint: {agentBaseUrl}");
             builder = builder.WithEnvironment("OTEL_EXPORTER_OTLP_ENDPOINT", agentBaseUrl);
             builder = builder.WithEnvironment("OTEL_METRIC_EXPORT_INTERVAL", "1000");
-            builder = builder.WithEnvironment("OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS", "AspNet");
+        }
+
+        foreach (var env in EnvironmentHelper.CustomEnvironmentVariables)
```
[the remark this answers is not in the document]
Comment (RassK) [reply test/IntegrationTests/Helpers/TestHelper.cs:95]: it's `CustomEnvironmentVariables`, the ones what ever you add extra for specific tests.

[the remark this answers is not in the document]
Comment (pellared) [reply test/IntegrationTests/Helpers/TestHelper.cs:95]: this comment is not about this change but the fact that `TestHelper.StartContainerAsync ` is used only [here](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/blob/e013026f8bea18de9e143b1ad3d2ccb2faba41bb/test/IntegrationTests/AspNetTests.cs#L55) so probably it could be moved to `AspNetTests.cs`

File: src/OpenTelemetry.AutoInstrumentation/Util/AssemblyDetector.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (pellared) [reply src/OpenTelemetry.AutoInstrumentation/Util/AssemblyDetector.cs:34]: I prefer to pass some references instead of having too much static code. This can be addressed in a separate PR.

File: docs/README.md [hunk 3]
```diff
@@ -145,7 +145,7 @@ OTEL_RESOURCE_ATTRIBUTES=deployment.environment=staging,service.version=1.0.0
 
 Enable desired library instrumentations using `OTEL_DOTNET_AUTO_TRACES_ENABLED_INSTRUMENTATIONS`
 and `OTEL_DOTNET_AUTO_METRICS_ENABLED_INSTRUMENTATIONS` environment variables.
-NOTE: By default all available **trace** instrumentations are included automatically.
+NOTE: By default all available instrumentations are included automatically.
```
[the remark this answers is not in the document]
Comment (RassK) [reply docs/README.md:148]: I'm basically fine with any outcome. Just to me it seems more important than for eg timeout config, as you actually can turn off quite big chunks.

**Review by pellared — APPROVED** [review]

File: docs/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (nrcventura) [reply docs/README.md:148]: I think that it makes sense to remove them from the README. By removing them from the README, it makes the configuration look similar to the way other vendor-specific offerings behave. But that can be done is a separate PR.

File: test/IntegrationTests/Helpers/TestHelper.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (nrcventura) [reply test/IntegrationTests/Helpers/TestHelper.cs:95]: I do not know how often we will want to run a test app in a container. It seems like an edge case to me. I think that it makes sense to move a lot of that logic for now into `AspNetTests.cs`. Doing so will make it easier for newcomers to know that structuring a test app in that manner is for special circumstances only. I think that this can be done in a separate PR.

File: src/OpenTelemetry.AutoInstrumentation/Util/AssemblyDetector.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (nrcventura) [reply src/OpenTelemetry.AutoInstrumentation/Util/AssemblyDetector.cs:34]: You could also pass the `AssemblyDetector` into the `EnvironmentConfiguration*Helper` methods as a parameter just like we do with the Settings object for some of the other methods. I think that the current approach is good enough for now, but a refactor could be explored in another PR.

File: test/IntegrationTests/AspNetTests.cs [hunk 4]
```diff
@@ -74,6 +74,9 @@ public async Task SubmitsTraces()
     [Trait("Containers", "Windows")]
     public async Task SubmitMetrics()
     {
+        // Helps to reduce noice by enabling only AspNet metrics.
```
Comment (nrcventura) [inline test/IntegrationTests/AspNetTests.cs:77]:
```suggestion
        // Helps to reduce noise by enabling only AspNet metrics.
```

**Review by nrcventura — APPROVED** [review]
I only noticed 1 minor misspelling, but otherwise this looks good to me.

**Review by rajkumar-rangaraj — APPROVED** [review]
